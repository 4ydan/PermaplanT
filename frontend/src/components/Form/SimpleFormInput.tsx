import { HTMLInputTypeAttribute } from 'react';
import { FieldValues, Path, UseFormRegister } from 'react-hook-form';

interface SimpleFormInputProps<T extends FieldValues> {
  id: Path<T>;
  labelText: string;
  placeHolder?: string;
  isArea?: boolean;
  required?: boolean;
  type?: HTMLInputTypeAttribute;
  defaultValue?: string | number | readonly string[];
  min?: number;
  max?: number;
  register?: UseFormRegister<T>;
  onChange?: () => void;
  valueAsNumber?: boolean;
  errorTitle?: string;
  disabled?: boolean;
  value?: string;
}

export default function SimpleFormInput<T extends FieldValues>({
  labelText,
  placeHolder = '',
  required = false,
  isArea = false,
  type = 'text',
  defaultValue,
  min,
  max,
  id,
  register,
  onChange,
  valueAsNumber = false,
  errorTitle,
  disabled = false,
  value,
}: SimpleFormInputProps<T>) {
  return (
    <div>
      <label htmlFor={id} className="mb-2 block text-sm font-medium">
        {labelText}
        {required ? <span className="text-red-800"> *</span> : <></>}
      </label>
      {isArea ? (
        <textarea
          disabled={disabled}
          onKeyUp={onChange}
          rows={6}
          name={id}
          id={id}
          className="block w-full rounded-lg border border-zinc-800 bg-primary-textfield dark:bg-primary-textfield-dark p-2.5 text-sm placeholder-neutral-700 focus:border-gray-600 focus:outline-none"
          placeholder={placeHolder}
          required={required}
          {...register?.(id)}
        />
      ) : (
        <input
          disabled={disabled}
          onKeyUp={onChange}
          type={type}
          min={min}
          max={max}
          id={id}
          defaultValue={defaultValue}
          className="block h-11 w-full rounded-lg border border-zinc-800 bg-primary-textfield dark:bg-primary-textfield-dark p-2.5 text-sm placeholder-neutral-700 focus:border-gray-600 focus:outline-none"
          placeholder={placeHolder}
          required={required}
          style={{ colorScheme: 'dark' }}
          pattern={valueAsNumber ? '^[0-9]+([,][0-9]{1,2})?$' : undefined}
          title={errorTitle}
          value={value?.toString()}
          {...register?.(id, {
            valueAsNumber: type === 'number' || valueAsNumber,
          })}
        />
      )}
    </div>
  );
}
