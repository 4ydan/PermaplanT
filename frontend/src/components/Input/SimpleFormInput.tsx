import { FieldValues, Path, UseFormRegister } from 'react-hook-form';
import { FormEvent, HTMLInputTypeAttribute } from 'react';

interface SimpleFormInputProps<T extends FieldValues> {
  id: Path<T>;
  labelText: string;
  placeHolder: string;
  isArea?: boolean;
  required?: boolean;
  type?: HTMLInputTypeAttribute;
  register?: UseFormRegister<T>;
  handleInputChange?: (event: FormEvent<HTMLInputElement>) => void;
  handleTextAreaChange?: (event: FormEvent<HTMLTextAreaElement>) => void;
}

export default function SimpleFormInput<T extends FieldValues>({
  labelText,
  placeHolder,
  required = false,
  isArea = false,
  type = 'text',
  id,
  register,
  handleInputChange,
  handleTextAreaChange,
}: SimpleFormInputProps<T>) {
  return (
    <div>
      <label htmlFor={id} className="mb-2 block text-sm font-medium  text-white">
        {labelText}
        {required ? <span className="text-red-800"> *</span> : <></>}
      </label>
      {isArea ? (
        <textarea
          name={id}
          id={id}
          className="block w-full rounded-lg border border-zinc-800 bg-primary-textfield p-2.5 text-sm text-white placeholder-neutral-700 focus:border-gray-600 focus:outline-none"
          placeholder={placeHolder}
          required={required}
          onChange={handleTextAreaChange}
        />
      ) : (
        <input
          type={type}
          id={id}
          className="block h-11 w-full rounded-lg border border-zinc-800 bg-primary-textfield p-2.5 text-sm text-white placeholder-neutral-700 focus:border-gray-600 focus:outline-none"
          placeholder={placeHolder}
          required={required}
          onChange={handleInputChange}
        />
      )}
    </div>
  );
}
