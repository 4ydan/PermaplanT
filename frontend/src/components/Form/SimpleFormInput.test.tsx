import renderer from 'react-test-renderer';
import SimpleFormInput from './SimpleFormInput';
import {MemoryRouter} from 'react-router-dom';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                    <SimpleFormInput id="test" labelText="Some Label" />
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});