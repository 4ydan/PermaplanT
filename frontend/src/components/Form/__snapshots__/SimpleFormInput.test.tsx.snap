// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dark:text-white"
>
  <label
    className="mb-2 block text-sm font-medium"
    htmlFor="test"
  >
    Some Label
  </label>
  <input
    className="block h-11 w-full rounded-lg border border-neutral-500 bg-neutral-100 p-2.5 text-sm placeholder-neutral-500 focus:border-primary-500 focus:outline-none aria-[invalid=true]:border-red-400 dark:border-neutral-400-dark dark:bg-neutral-50-dark dark:focus:border-primary-300 aria-[invalid=true]:dark:border-red-400"
    id="test"
  />
</div>
`;
